readthedocs/builds/version_slug.py

```diff
@@ -114,13 +104,40 @@ def slugify(self, content):
             space_replacement='-',
         )
 
-        # Remove first character wile it's an invalid character for the
+        # Remove first character while it's an invalid character for the
         # beginning of the slug
-        slugified = slugified.lstrip(self.ok_chars)
+        slug = slug.lstrip(self.ok_chars)
+        slug = slug or self.fallback_slug
+
+        if not self.is_valid(slug):
+            return None
+        return slug
+
 
-        if not slugified:
-            return self.fallback_slug
-        return slugified
+class VersionSlugField(models.CharField):
```

Comment:
Why did these files change? There's nothing in the PR description about what logic changes are here.